api/src/main/java/org/openmrs/CachingConfig.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import net.sf.ehcache.config.CacheConfiguration;
+import org.apache.commons.beanutils.BeanUtils;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.CachingConfigurer;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.cache.ehcache.EhCacheCacheManager;
+import org.springframework.cache.ehcache.EhCacheManagerFactoryBean;
+import org.springframework.cache.interceptor.CacheErrorHandler;
+import org.springframework.cache.interceptor.CacheResolver;
+import org.springframework.cache.interceptor.KeyGenerator;
+import org.springframework.cache.interceptor.SimpleCacheErrorHandler;
+import org.springframework.cache.interceptor.SimpleCacheResolver;
+import org.springframework.cache.interceptor.SimpleKeyGenerator;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
+import org.springframework.core.io.support.ResourcePatternResolver;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+
+@Configuration
+@EnableCaching
+public class CachingConfig implements CachingConfigurer{
+
+    @Bean
+    public EhCacheManagerFactoryBean ehCacheManagerFactoryBean(){
+        EhCacheManagerFactoryBean cacheManagerFactoryBean = new EhCacheManagerFactoryBean();
+        cacheManagerFactoryBean.setConfigLocation(new ClassPathResource("ehcache-api.xml"));
+        cacheManagerFactoryBean.setShared(true);
+
+        return cacheManagerFactoryBean;
+    }
+
+    @Bean(name = "apiCacheManager")
+    public CacheManager apiCacheManager(){
+       return new EhCacheCacheManager(apiEhCacheManager());
+    }
+
+    @Bean(destroyMethod = "shutdown", name = "apiEhCacheManager")
+    public net.sf.ehcache.CacheManager apiEhCacheManager(){
+
+        net.sf.ehcache.config.Configuration config = ehCacheManagerFactoryBean().getObject().getConfiguration();
+
+        List<OpenmrsCacheConfiguration> cacheConfigurations = getCacheConfigurations();
+        cacheConfigurations.stream()
```

Comment:
Also it suggests to extend  CachingConfigurerSupport instead of implementing CachingConfigurer